diff --git a/tests/misc/tsort.pl b/tests/misc/tsort.pl
index 70bdc474c..4fd420a4e 100755
--- a/tests/misc/tsort.pl
+++ b/tests/misc/tsort.pl
@@ -54,8 +54,10 @@ my @Tests =
 
    ['only-one', {IN => {f => ""}}, {IN => {g => ""}},
     {EXIT => 1},
-    {ERR => "tsort: extra operand 'g'\n"
-     . "Try 'tsort --help' for more information.\n"}],
+    {ERR => "error: unexpected argument 'g' found\n\n"
+         . "Usage: tsort [OPTIONS] FILE\n\n"
+         . "For more information, try '--help'.\n"
+    }],
   );
 
 my $save_temps = $ENV{DEBUG};
